File: spotify_clone_app/lib/features/auth/dtos/refresh_token_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'refresh_token_dto.g.dart';

@JsonSerializable()
class RefreshTokenDto {
  final int userId;
  final String refreshToken;
  final String deviceId;

  const RefreshTokenDto({
    required this.userId,
    required this.refreshToken,
    required this.deviceId,
  });

  Map<String, dynamic> toJson() => _$RefreshTokenDtoToJson(this);

  factory RefreshTokenDto.fromJson(Map<String, dynamic> json) => _$RefreshTokenDtoFromJson(json);
}

File: spotify-backend/src/auth/dtos/refresh-token.dto.ts
import { IsNotEmpty, IsInt, IsString } from 'class-validator';

export class RefreshTokenDto {
  @IsInt()
  @IsNotEmpty()
  userId: number;

  @IsString()
  @IsNotEmpty()
  refreshToken: string;

  @IsString()
  @IsNotEmpty()
  deviceId: string;
}